hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -542,7 +547,8 @@ SqlResult execute(SelectPlan plan,
         try {
             sqlJobInvocationObservers.forEach(observer -> observer.onJobInvocation(plan.getDag(), jobConfig));
             Job job = plan.isAnalyzed()
-                    ? jet.newJob(jobId, plan.getDag(), jobConfig, ssc.subject())
+                    // Note: identifier in programming languages cannot be started from digit.
+                    ? jet.newJob(jobId, plan.getDag(), jobConfig.setName("x" + jobId), ssc.subject())
```

Comment:
job name is not mandatory so why generate it?

Reply:
see my rationale in https://hazelcast.atlassian.net/browse/HZ-3586?focusedCommentId=88999